Demo/Demo/SwiftUIComponents/PayPalWebViews/PayPalOrderActionButton.swift

```diff
@@ -0,0 +1,53 @@
+import SwiftUI
+
+struct PayPalOrderActionButton: View {
+
+    let intent: Intent
+    let orderID: String
+    let selectedMerchantIntegration: MerchantIntegration
+
+    @ObservedObject var paypalWebViewModel: PayPalWebViewModel
+
+    var body: some View {
+        ZStack {
+            Button("\(intent.rawValue)") {
+                completeOrder()
+            }
+            .buttonStyle(RoundedBlueButtonStyle())
+            .padding()
+
+            if .loading == paypalWebViewModel.state.authorizedOrderResponse ||
+                .loading == paypalWebViewModel.state.capturedOrderResponse {
+                CircularProgressView()
+            }
+        }
+    }
+
+    private func completeOrder() {
+        if intent == .capture {
+            Task {
+                do {
+                    try await paypalWebViewModel.captureOrder(
```

Comment:
It seems like both this and the else have basically the same code repeated. Can we pass the intent to the view model to dry up this code? Something like:
```swift
try await paypalWebViewModel(with: intent, orderID: orderID, selectedMerchantIntegration: selectedMerchantIntegration)
```
Then in the view model we can do a switch on the intent passed in.

Reply:
I have the intent toggle on the main screen for PayPalWebPayment. It would not make sense imo to instantiate the PayPalWEbViewModel with intent option set.

Reply:
We aren't instantiating the view model here we are calling a method called `paypalWebViewModel.captureOrder/authroizeOrder`. The suggestion is to combine the logic here into 1 method on  `paypalWebViewModel`. It can be called whatever we want, so if order is more clear we can do `try await paypalWebViewModel.order(with: intent, orderID: orderID, selectedMerchantIntegration: selectedMerchantIntegration)`. I just removed the method name `captureOrder/authroizeOrder` to illustrate the idea.

Reply:
That makes sense. I will have one function that redirects from the viewModel. Thank you!